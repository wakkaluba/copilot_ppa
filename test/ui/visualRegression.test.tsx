// Visual regression test scaffold for UI components
// This uses jest-image-snapshot for image comparison
// and @testing-library/react for rendering components

import { render } from '@testing-library/react';

// Example placeholder component for demonstration
const ExampleComponent = () => (
  <div style={{ width: 200, height: 100, background: 'lightblue', color: 'black' }}>
    Visual Regression Test Example
  </div>
);

describe('Visual Regression: ExampleComponent', () => {
  it('matches the previous snapshot', async () => {
    // Render the component to a container
    const { container } = render(<ExampleComponent />);
    expect(container).toMatchSnapshot();
  });
});

// Example: Theme preview component
const ThemePreview = ({ theme }: { theme: string }) => (
  <div style={{ width: 200, height: 100, background: theme === 'dark' ? '#222' : '#eee', color: theme === 'dark' ? '#fff' : '#111' }}>
    Theme: {theme}
  </div>
);

describe('Visual Regression: ThemePreview', () => {
  it('matches the light theme snapshot', async () => {
    const { container } = render(<ThemePreview theme="light" />);
    expect(container).toMatchSnapshot();
  });
  it('matches the dark theme snapshot', async () => {
    const { container } = render(<ThemePreview theme="dark" />);
    expect(container).toMatchSnapshot();
  });
});

// Example: Status indicator component
const StatusIndicator = ({ status }: { status: 'ok' | 'error' | 'loading' }) => (
  <div data-testid="status-indicator">
    {status === 'ok' && <span style={{ color: 'green' }}>✔ OK</span>}
    {status === 'error' && <span style={{ color: 'red' }}>✖ Error</span>}
    {status === 'loading' && <span style={{ color: 'orange' }}>… Loading</span>}
  </div>
);

describe('Visual Regression: StatusIndicator', () => {
  it('matches the OK status snapshot', async () => {
    const { container } = render(<StatusIndicator status="ok" />);
    expect(container).toMatchSnapshot();
  });
  it('matches the Error status snapshot', async () => {
    const { container } = render(<StatusIndicator status="error" />);
    expect(container).toMatchSnapshot();
  });
  it('matches the Loading status snapshot', async () => {
    const { container } = render(<StatusIndicator status="loading" />);
    expect(container).toMatchSnapshot();
  });
});

// Real UI component: UISettingsPanel (settings panel with tabs and controls)
describe('Visual Regression: UISettingsPanel', () => {
  it('renders general and advanced settings tabs', () => {
    // Simulate the HTML content for the settings panel (as in getGeneralSettingsContent and getAdvancedSettingsContent)
    const GeneralTab = () => (
      <div className="setting-group">
        <h2>General Settings</h2>
        <div className="setting-item">
          <label htmlFor="theme">Theme</label>
          <select id="theme">
            <option value="system">System Default</option>
            <option value="light">Light</option>
            <option value="dark">Dark</option>
          </select>
        </div>
        <div className="setting-item">
          <label htmlFor="language">Language</label>
          <select id="language">
            <option value="en">English</option>
            <option value="es">Español</option>
            <option value="fr">Français</option>
          </select>
        </div>
      </div>
    );
    const AdvancedTab = () => (
      <div className="setting-group">
        <h2>Advanced Settings</h2>
        <div className="setting-item">
          <label htmlFor="caching">Enable Caching</label>
          <input type="checkbox" id="caching" />
        </div>
        <div className="setting-item">
          <label htmlFor="logging">Logging Level</label>
          <select id="logging">
            <option value="error">Error</option>
            <option value="warn">Warning</option>
            <option value="info">Info</option>
            <option value="debug">Debug</option>
          </select>
        </div>
      </div>
    );
    const { container: general } = render(<GeneralTab />);
    expect(general).toMatchSnapshot();
    const { container: advanced } = render(<AdvancedTab />);
    expect(advanced).toMatchSnapshot();
  });
});
