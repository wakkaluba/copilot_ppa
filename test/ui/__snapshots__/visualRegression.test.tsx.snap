// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visual Regression: ExampleComponent matches the previous snapshot 1`] = `
<div>
  <div
    style="width: 200px; height: 100px; background: lightblue; color: black;"
  >
    Visual Regression Test Example
  </div>
</div>
`;

exports[`Visual Regression: StatusIndicator matches the Error status snapshot 1`] = `
<div>
  <div
    data-testid="status-indicator"
  >
    <span
      style="color: red;"
    >
      ✖ Error
    </span>
  </div>
</div>
`;

exports[`Visual Regression: StatusIndicator matches the Loading status snapshot 1`] = `
<div>
  <div
    data-testid="status-indicator"
  >
    <span
      style="color: orange;"
    >
      … Loading
    </span>
  </div>
</div>
`;

exports[`Visual Regression: StatusIndicator matches the OK status snapshot 1`] = `
<div>
  <div
    data-testid="status-indicator"
  >
    <span
      style="color: green;"
    >
      ✔ OK
    </span>
  </div>
</div>
`;

exports[`Visual Regression: ThemePreview matches the dark theme snapshot 1`] = `
<div>
  <div
    style="width: 200px; height: 100px; background: rgb(34, 34, 34); color: rgb(255, 255, 255);"
  >
    Theme: 
    dark
  </div>
</div>
`;

exports[`Visual Regression: ThemePreview matches the light theme snapshot 1`] = `
<div>
  <div
    style="width: 200px; height: 100px; background: rgb(238, 238, 238); color: rgb(17, 17, 17);"
  >
    Theme: 
    light
  </div>
</div>
`;

exports[`Visual Regression: UISettingsPanel renders general and advanced settings tabs 1`] = `
<div>
  <div
    class="setting-group"
  >
    <h2>
      General Settings
    </h2>
    <div
      class="setting-item"
    >
      <label
        for="theme"
      >
        Theme
      </label>
      <select
        id="theme"
      >
        <option
          value="system"
        >
          System Default
        </option>
        <option
          value="light"
        >
          Light
        </option>
        <option
          value="dark"
        >
          Dark
        </option>
      </select>
    </div>
    <div
      class="setting-item"
    >
      <label
        for="language"
      >
        Language
      </label>
      <select
        id="language"
      >
        <option
          value="en"
        >
          English
        </option>
        <option
          value="es"
        >
          Español
        </option>
        <option
          value="fr"
        >
          Français
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`Visual Regression: UISettingsPanel renders general and advanced settings tabs 2`] = `
<div>
  <div
    class="setting-group"
  >
    <h2>
      Advanced Settings
    </h2>
    <div
      class="setting-item"
    >
      <label
        for="caching"
      >
        Enable Caching
      </label>
      <input
        id="caching"
        type="checkbox"
      />
    </div>
    <div
      class="setting-item"
    >
      <label
        for="logging"
      >
        Logging Level
      </label>
      <select
        id="logging"
      >
        <option
          value="error"
        >
          Error
        </option>
        <option
          value="warn"
        >
          Warning
        </option>
        <option
          value="info"
        >
          Info
        </option>
        <option
          value="debug"
        >
          Debug
        </option>
      </select>
    </div>
  </div>
</div>
`;
